engine/src/main/java/org/apache/apex/engine/YarnAppLauncherImpl.java

```diff
@@ -124,13 +121,15 @@ private void setConfiguration(Configuration conf, String property, Object value)
     }
   }
 
-  public class YarnAppHandleImpl implements YarnAppLauncher.YarnAppHandle
+  public class YarnAppHandleImpl implements YarnAppLauncher.YarnAppHandle, AutoCloseable
   {
     final ApplicationId appId;
+    final YarnClient yarnClient;
```

Comment:
can visibility be limited to private? Do you want to expose yarnClient outside? Somebody may call yarnClient.stop() invalidating YarnAppHandleImpl  state.